src/caffe/layers/data_layer.cpp

```diff
@@ -74,6 +74,7 @@ void DataLayer<Dtype>::Next() {
         << "Restarting data prefetching from start.";
     cursor_->SeekToFirst();
   }
+  offset_++;
```

Comment:
It's on purpose, the round robin can continue the same way, even if the database has to reset. It's easier to think of it on the unrolled dataset.